src/Microsoft.DotNet.Darc/src/DarcLib/GitRepoCloner.cs

```diff
@@ -0,0 +1,178 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.DarcLib.Helpers;
+using System.IO;
+using System.Linq;
+using System;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+namespace Microsoft.DotNet.DarcLib;
+
+public class GitRepoCloner : IGitRepoCloner
+{
+    private readonly ILogger _logger;
+    private readonly string _personalAccessToken;
+
+    public GitRepoCloner(string personalAccessToken, ILogger logger)
+    {
+        _logger = logger;
+        _personalAccessToken = personalAccessToken;
+    }
+
+    public void Clone(string repoUri, string commit, string targetDirectory, bool checkoutSubmodules, string? gitDirectory = null)
+    {
+        Clone(repoUri, commit, targetDirectory, checkoutSubmodules, _logger, _personalAccessToken, gitDirectory);
+    }
+
+    /// <summary>
+    ///     Clone a remote git repo.
+    /// </summary>
+    /// <param name="repoUri">Repository uri to clone</param>
+    /// <param name="commit">Branch, commit, or tag to checkout</param>
+    /// <param name="targetDirectory">Target directory to clone to</param>
+    /// <param name="checkoutSubmodules">Indicates whether submodules should be checked out as well</param>
+    /// <param name="gitDirectory">Location for the .git directory, or null for default</param>
+    /// <returns></returns>
+    protected static void Clone(string repoUri, string commit, string targetDirectory, bool checkoutSubmodules, ILogger logger, string pat, string? gitDirectory)
```

Comment:
Is this `Clone` now called from any of the child classes? If not, we could merge it with the method above